src/com/Jacksonnn/DCCore/QuickDeposit/QuickDepositListener.java

```diff
@@ -11,49 +12,14 @@
 import org.bukkit.event.player.PlayerInteractEvent;
 import org.bukkit.inventory.ItemStack;
 
-import static com.Jacksonnn.DCCore.Configuration.ConfigManager.defaultConfig;
-
 public class QuickDepositListener implements Listener {
 
-	public static boolean qdEnabled = false;
-
-	@EventHandler
-	public void onLeftClick(PlayerInteractEvent event) {
-		if (defaultConfig.get().getBoolean("QuickDeposit.players." + event.getPlayer().getName())) {
-			if (event.getPlayer().hasPermission("DCCore.QuickDeposit.use")) {
-				if (event.getPlayer().isSneaking()) {
-					if (event.getAction() == Action.LEFT_CLICK_BLOCK) {
-						BlockState state = event.getClickedBlock().getState();
-
-						if (state instanceof Chest) {
-							Chest chest = (Chest) state;
-							Material blockAbove = event.getClickedBlock().getRelative(BlockFace.UP).getType();
-							if (blockAbove.isOccluding()) {
-								return;
-							}
-							int empty = chest.getBlockInventory().firstEmpty();
-
-							if (empty != -1) {
-								ItemStack handItem = event.getPlayer().getInventory().getItemInMainHand();
-
-								chest.getBlockInventory().addItem(handItem);
-
-								event.getPlayer().getInventory().removeItem(handItem);
-								event.getPlayer().updateInventory();
-							}
-						}
-					}
-				}
-			}
-		}
-	}
-	
 	@EventHandler
 	public void onRightClick(PlayerInteractEvent event) {
 
 		Player player = event.getPlayer();
 
-		if (qdEnabled) {
+		if (GeneralMethods.getDCPlayer(event.getPlayer().getUniqueId()).isQuickdeposit()) {
```

Comment:
Consider ensuring that GeneralMethods.getDCPlayer() does not equal null before calling .isQuickdeposit()